test/blackbox-tests/test-cases/pkg/opam-package-with-patch-filter.t

```diff
@@ -0,0 +1,61 @@
+We test the translation of an opam package with a patches field with a filter into a dune
+lock file.
+
+  $ . ./helpers.sh
+
+Generate a mock opam repository
+  $ mkdir -p mock-opam-repository
+  $ cat >mock-opam-repository/repo <<EOF
+  > opam-version: "2.0"
+  > EOF
+
+Make a package with a patch behind a filter
+  $ mkpkg with-patch-filter <<EOF
+  > opam-version: "2.0"
+  > patches: ["foo.patch" {with-test}]
```

Comment:
@rgrinberg I've removed the `os` stuff that isn't in opam and replaced it with `with-test` that I am sure is always here. Since we don't build the package in this mode, it should work when we add the filters.

Reply:
The problem with this test is that it's not obvious that you're solving with the test filter turned. Why don't you use os instead?

Reply:
Does it matter what `with-test` is when we solve? It should always give a `when` anyway.

Reply:
Oh yeah, we only evaluate filters for dependencies.